File: backend/models/api/tv.go
package api

import (
	dbModels "github.com/STREAM-BUSTER/stream-buster/models/db"
	"time"
)

type TV struct {
	MediaID uint
	Media   *dbModels.Media

	SeasonCount  int
	EpisodeCount int

	FirstAirDate  *time.Time
	LastAirDate   *time.Time
	BackdropImage string

	Seasons []*Season
}

File: frontend/src/models/TV.ts
import { Media } from './Media';
import { Season } from './Season';  // Assuming a Season interface

export interface TV {
  mediaID: number;
  media?: Media;  // Optional field
  overview: string;
  seasonCount: number;
  episodeCount: number;
  seasons: Season[];  // Array of Seasons
  firstAirDate?: Date;  // Optional field
}